### PR TITLE
Emby.Server.Implementations code cleanup and warning fixes (Part 2)

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
Documents and cleans up code in Emby.Server.Implementations.Data.

This is the second part of a series of PR's that hopefully will eliminate all warnings in the Emby.Server.Implementations project.

**Related Issues**
#2149

### CODE REVIEW

File: Emby.Server.Implementations/Data/SqliteDisplayPreferencesRepository.cs [hunk 1]
```diff
@@ -168,19 +167,15 @@ public DisplayPreferences GetDisplayPreferences(string displayPreferencesId, Gui
 
             var guidId = displayPreferencesId.GetMD5();
 
-            using (var connection = GetConnection(true))
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteDisplayPreferencesRepository.cs:171]: Why did you remove this argument?

Comment (barronpm) [reply Emby.Server.Implementations/Data/SqliteDisplayPreferencesRepository.cs:171]: `GetConnection` never actually uses the argument, so I removed the argument to make it easier in the future to remove that parameter from the method. I was originally going to do it in this PR, but it would've required me to make changes that would conflict with part 1, so I'm saving it until later.

Comment (Bond-009) [reply Emby.Server.Implementations/Data/SqliteDisplayPreferencesRepository.cs:171]: I left the parameter in purpose, it used to determine if a read only connection should be returned or a read-write connection. By leaving it in we can easily add that back in the future.

Comment (barronpm) [reply Emby.Server.Implementations/Data/SqliteDisplayPreferencesRepository.cs:171]: Alright, I'll add it back.

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 2]
```diff
@@ -4927,23 +4582,12 @@ public List<PersonInfo> GetPeople(InternalPeopleQuery query)
...
-                    }
-                }
+            using var connection = GetConnection();
+            using var statement = PrepareStatement(connection, commandText);
+            // Run this again to bind the params
+            GetPeopleWhereClauses(query, statement);
 
-                return list;
-            }
+            return statement.ExecuteQuery().Select(row => GetPerson(row)).ToList();
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:4590]:
```suggestion
            return statement.ExecuteQuery().Select(GetPerson).ToList();
```

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 3]
```diff
@@ -5154,19 +4788,14 @@ private List<string> GetItemValueNames(int[] itemValueTypes, List<string> withIt
             commandText += " Group By CleanValue";
 
             var list = new List<string>();
-            using (var connection = GetConnection(true))
+            using var connection = GetConnection();
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:4791]: Do not remove parameter here

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 4]
```diff
@@ -5341,86 +4970,82 @@ private QueryResult<(BaseItem, ItemCounts)> GetItemValues(InternalItemsQuery que
 
             var list = new List<(BaseItem, ItemCounts)>();
             var result = new QueryResult<(BaseItem, ItemCounts)>();
-            using (var connection = GetConnection(true))
-            {
-                connection.RunInTransaction(
-                    db =>
-                    {
-                        var statements = PrepareAll(db, statementTexts).ToList();
+            using var connection = GetConnection();
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:4973]: Do not remove parameter here

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 5]
```diff
@@ -5720,34 +5340,25 @@ public List<MediaStream> GetMediaStreams(MediaStreamQuery query)
...
+                statement.TryBind("@StreamType", query.Type.Value.ToString());
+            }
 
-                return list;
+            if (query.Index.HasValue)
+            {
+                statement.TryBind("@StreamIndex", query.Index.Value);
             }
+
+            return statement.ExecuteQuery().Select(row => GetMediaStream(row)).ToList();
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:5358]:
```suggestion
            return statement.ExecuteQuery().Select(GetMediaStream).ToList();
```

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 6]
```diff
@@ -6073,26 +5679,19 @@ public List<MediaAttachment> GetMediaAttachments(MediaAttachmentQuery query)
...
-                {
-                    list.Add(GetMediaAttachment(row));
-                }
+            if (query.Index.HasValue)
+            {
+                statement.TryBind("@AttachmentIndex", query.Index.Value);
             }
 
-            return list;
+            return statement.ExecuteQuery().Select(row => GetMediaAttachment(row)).ToList();
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:5691]:
```suggestion
            return statement.ExecuteQuery().Select(GetMediaAttachment).ToList();
```

File: Emby.Server.Implementations/Data/BaseSqliteRepository.cs [hunk 7]
```diff
@@ -9,9 +7,12 @@
 
 namespace Emby.Server.Implementations.Data
 {
+    /// <summary>
+    /// The base Sqlite repository class.
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/BaseSqliteRepository.cs:11]:
```suggestion
    /// The base SQLite repository class.
```

File: Emby.Server.Implementations/Data/BaseSqliteRepository.cs [hunk 8]
```diff
@@ -96,6 +97,10 @@ protected BaseSqliteRepository(ILogger logger)
         /// <value>The write connection.</value>
         protected SQLiteDatabaseConnection WriteConnection { get; set; }
 
+        /// <summary>
+        /// Gets a connection to the database.
+        /// </summary>
+        /// <returns>A <see cref="ManagedConnection"/>.</returns>
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/BaseSqliteRepository.cs:103]: Should document the return value, not the type

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 9]
```diff
@@ -1170,9 +1154,7 @@ public ItemImageInfo ItemImageInfoFromValueString(string value)
                 return null;
             }
 
-            var image = new ItemImageInfo();
-
-            image.Path = RestorePath(parts[0]);
+            var image = new ItemImageInfo { Path = RestorePath(parts[0]) };
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:1157]: Multiple lines

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 10]
```diff
@@ -2636,59 +2533,50 @@ public List<BaseItem> GetItemList(InternalItemsQuery query)
                 }
             }
 
-            var items = new List<BaseItem>();
-            using (var connection = GetConnection(true))
+            using var connection = GetConnection();
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:2536]:
```suggestion
            using var connection = GetConnection(true);
```
Please check again in case you missed more

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 11]
```diff
@@ -3319,9 +3189,9 @@ private bool IsAlphaNumeric(string str)
                 return false;
             }
 
-            for (int i = 0; i < str.Length; i++)
+            foreach (var t in str)
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:3192]: Not sure about this change, maybe the `for` loop was used for performance

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 12]
```diff
@@ -4890,24 +4550,20 @@ public List<string> GetPeopleNames(InternalPeopleQuery query)
...
-                    foreach (var row in statement.ExecuteQuery())
-                    {
-                        list.Add(row.GetString(0));
-                    }
-                }
+                // Run this again to bind the params
+                GetPeopleWhereClauses(query, statement);
 
-                return list;
+                list.AddRange(statement.ExecuteQuery().Select(row => row.GetString(0)));
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:4560]: Why use `AddRange` here? `return .. .ToList()`

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 13]
```diff
@@ -5726,34 +5347,25 @@ public List<MediaStream> GetMediaStreams(MediaStreamQuery query)
...
-
-                using (var statement = PrepareStatement(connection, cmdText))
-                {
-                    statement.TryBind("@ItemId", query.ItemId.ToByteArray());
-
-                    if (query.Type.HasValue)
-                    {
-                        statement.TryBind("@StreamType", query.Type.Value.ToString());
-                    }
+            using var connection = GetConnection();
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:5350]:
```suggestion
            using var connection = GetConnection(true);
```

File: Emby.Server.Implementations/Data/SqliteUserDataRepository.cs [hunk 14]
```diff
@@ -343,9 +307,8 @@ public List<UserItemData> GetAllUserData(long internalUserId)
         /// <param name="reader"></param>
         private UserItemData ReadRow(IReadOnlyList<IResultSetValue> reader)
         {
-            var userData = new UserItemData();
+            var userData = new UserItemData { Key = reader[0].ToString() };
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteUserDataRepository.cs:310]: Use multiple lines

**Comment by barronpm** [comment]
Closing this PR as these files are being removed and fixing the issues with this PR is no longer worth it.
